### PR TITLE
Increase the stack size

### LINKED ISSUE #6319: Backend crash on startup
In current `develop` (b700679d35), on Linux, I am unable to start the application (`./run ide start --skip-wasm-opt`):
```
[...]
 INFO ide_ci::program::command: enso⚠️ [info] [2023-04-17T18:21:44.549Z] [org.enso.projectmanager.boot.ProjectManager$] Starting Project Manager...
 INFO ide_ci::program::command: enso⚠️ [info] [2023-04-17T18:21:44.567Z] [org.enso.projectmanager.boot.ProjectManager$] Started server at 127.0.0.1:30535, press enter to kill server
 INFO ide_ci::program::command: enso⚠️ [info] [2023-04-17T18:21:46.018Z] [org.enso.projectmanager.protocol.ClientController] Client connected to Project Manager [c5402c02-8675-4383-a3f2-4a11a0ad5af2]
 INFO ide_ci::program::command: enso⚠️ java.lang.StackOverflowErrorjava.lang.StackOverflowErrorjava.lang.StackOverflowErrorjava.lang.StackOverflowErrorjava.lang.StackOverflowError
 INFO ide_ci::program::command: enso⚠️   at org.graalvm.nativeimage.builder/com.oracle.svm.core.graal.snippets.StackOverflowCheckImpl.newStackOverflowError0(StackOverflowCheckImpl.java:333)
 INFO ide_ci::program::command: enso⚠️   at org.graalvm.nativeimage.builder/com.oracle.svm.core.graal.snippets.StackOverflowCheckImpl.newStackOverflowError(StackOverflowCheckImpl.java:329)
 INFO ide_ci::program::command: enso⚠️   at org.graalvm.nativeimage.builder/com.oracle.svm.core.graal.snippets.StackOverflowCheckImpl.throwNewStackOverflowError(StackOverflowCheckImpl.java:309)
 INFO ide_ci::program::command: enso⚠️   at org.graalvm.nativeimage.builder/com.oracle.svm.core.jni.functions.JNIFunctions.FindClass(JNIFunctions.java:340)                                                                                                                                                                    
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.FileOutputStream.writeBytes(FileOutputStream.java)                                       
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.FileOutputStream.write(FileOutputStream.java:354)                                        
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:81)                           
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.BufferedOutputStream.write(BufferedOutputStream.java:127)                                                                                                                                                                                                
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.PrintStream.write(PrintStream.java:559)                                                                                                                                                                                                                  
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/sun.nio.cs.StreamEncoder.writeBytes(StreamEncoder.java:233)                                      
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/sun.nio.cs.StreamEncoder.implFlushBuffer(StreamEncoder.java:312)                                                                                                                                                                                                 
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/sun.nio.cs.StreamEncoder.flushBuffer(StreamEncoder.java:104)                                     
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.OutputStreamWriter.flushBuffer(OutputStreamWriter.java:181)                               
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.PrintStream.write(PrintStream.java:606)                                                  
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.PrintStream.print(PrintStream.java:745)                                                  
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.io.PrintStream.println(PrintStream.java:899)                                                
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.lang.Throwable$WrappedPrintStream.println(Throwable.java:752)                                                                                                                                                                                               
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.lang.Throwable.printStackTrace(Throwable.java:659)                                           
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.lang.Throwable.printStackTrace(Throwable.java:648)                                          
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.lang.Throwable.printStackTrace(Throwable.java:639)                                           
 INFO ide_ci::program::command: enso⚠️   at zio.RuntimePlatformSpecific.$anonfun$defaultReportFatal$1(RuntimePlatformSpecific.scala:38)                                                                                                                                                                                        
 INFO ide_ci::program::command: enso⚠️   at zio.internal.FiberRuntime.handleFatalError(FiberRuntime.scala:674)                                                                                                                                                                                                                 
 INFO ide_ci::program::command: enso⚠️   at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1274)                                                          
 INFO ide_ci::program::command: enso⚠️   at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:944)
[...previous line repeated many times...]
 INFO ide_ci::program::command: enso⚠️   at zio.internal.FiberRuntime.evaluateEffect(FiberRuntime.scala:391)
 INFO ide_ci::program::command: enso⚠️   at zio.internal.FiberRuntime.evaluateMessageWhileSuspended(FiberRuntime.scala:514)
 INFO ide_ci::program::command: enso⚠️   at zio.internal.FiberRuntime.drainQueueOnCurrentThread(FiberRuntime.scala:230)
 INFO ide_ci::program::command: enso⚠️   at zio.internal.FiberRuntime.run(FiberRuntime.scala:139)
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.util.concurrent.FutureTask.run(FutureTask.java:264)
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 INFO ide_ci::program::command: enso⚠️   at java.base@11.0.18/java.lang.Thread.run(Thread.java:829)
 INFO ide_ci::program::command: enso⚠️   at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:775)
 INFO ide_ci::program::command: enso⚠️   at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:203)
 INFO ide_ci::program::command: ensoℹ️ **** WARNING ****
 INFO ide_ci::program::command: ensoℹ️ Catastrophic error encountered. Application not safely interrupted. Resources may be leaked. Check the logs for more details and consider overriding `Runtime.reportFatal` to capture context.
 INFO ide_ci::program::command: ensoℹ️ Backend exited with code 255.
```

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #6319 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

**Review by hubertp — COMMENTED** [review]
@kazcw but you mentioned in https://github.com/enso-org/enso/issues/6319#issuecomment-1514778641 that it doesn't fix the problem for you. So... something changed?

**Comment by kazcw** [comment]
> @kazcw but you mentioned in [#6319 (comment)](https://github.com/enso-org/enso/issues/6319#issuecomment-1514778641) that it doesn't fix the problem for you. So... something changed?

I mentioned that [a6b6c69](https://github.com/enso-org/enso/pull/6355/commits/a6b6c69569ecbd48613a22302923404920ce8135) didn't fix it. The current version of this PR is working for me.

**Review by hubertp — APPROVED** [review]

File: project/NativeImage.scala [hunk 1]
```diff
@@ -32,22 +32,26 @@ object NativeImage {
...
     artifactName: String,
     staticOnLinux: Boolean,
-    additionalOptions: Seq[String]    = Seq.empty,
-    memoryLimitMegabytes: Option[Int] = Some(15608),
-    initializeAtRuntime: Seq[String]  = Seq.empty,
-    mainClass: Option[String]         = None,
-    cp: Option[String]                = None
+    additionalOptions: Seq[String]           = Seq.empty,
+    buildMemoryLimitMegabytes: Option[Int]   = Some(15608),
+    runtimeThreadStackMegabytes: Option[Int] = Some(2),
```
Comment (JaroslavTulach) [inline project/NativeImage.scala:51]: Runtime thread stack megabytes is the change? The default is 1MB, I assume. Shouldn't the default remain `1MB` and only in cases where we know we need two, we'd have to provide that explicitly?

Comment (4e6) [reply project/NativeImage.scala:51]: The comment says that 2M is the minimal amount required for ZIO to work. I believe we can leave it as a default

**Pull request merged**